src/toffy/json_utils.py

```diff
@@ -240,3 +240,39 @@ def check_fov_resolutions(bin_file_dir, run_name, save_path=None):
         resolution_data.to_csv(save_path, index=False)
 
     return resolution_data
+
+
+def missing_fov_check(bin_file_dir, run_name):
+    """Use the run metadata to check if all FOV data was generated
+    Args:
+        bin_file_dir (str): directory containing the run json file
+        run_name (str): name of the run and corresponding run file
+
+    Raises:
+        FileNotFoundError: FOV specified in the run file doesn't have associated bin/json file
+    """
+
+    # read in run file
+    run_file_path = os.path.join(bin_file_dir, run_name + ".json")
+    io_utils.validate_paths([run_file_path])
+    run_metadata = read_json_file(run_file_path, encoding="utf-8")
+
+    missing_fovs = {}
+    for fov in run_metadata.get("fovs", ()):
+        # get fov names
+        fov_number = fov.get("runOrder")
+        default_name = f"fov-{fov_number}-scan-1"
+        custom_name = fov.get("name")
+
+        # add default name and custom name for missing data
+        if not os.path.exists(
+            os.path.join(bin_file_dir, default_name + ".bin")
+        ) or not os.path.exists(os.path.join(bin_file_dir, default_name + ".json")):
+            missing_fovs[default_name] = [custom_name]
+
+        elif os.path.getsize(os.path.join(bin_file_dir, default_name + ".json")) == 0:
```

Comment:
This will be problematic if both the `.bin` and `.json` file did get written out after the watcher timeout threshold. I think you should check the extracted image folder directory to see if the FOV folder actually exists; it might really be the only check you need.

Reply:
This is super helpful, thank you!

Reply:
In the edge case where a FOV times out but actually gets written afterwards, it will not be processed (have an extracted folder). It's probably best to keep the check as is, and warn the user of this processing mishap.